lib/http/request.rb

```diff
@@ -106,7 +106,7 @@ def redirect(uri, verb = @verb)
 
     # Stream the request to a socket
     def stream(socket)
-      include_proxy_headers if using_proxy? && !@uri.https?
```

Comment:
I believe that in a https connection this is the SSL stream, not the CONNECT request, so the proxy headers should have already been sent with the CONNECT (see proxy_connect_headers), not through the encrypted stream